### PR TITLE
feat(plugin_runner): shared runtime engine

### PR DESCRIPTION
**Pull request by kwonoj**
**Description:**

Attempt to address suggestion: https://github.com/wasmerio/wasmer/issues/3793#issuecomment-1607117480

### CODE REVIEW

**Comment by syrusakbary** [comment]
Note: this might not be needed once this is merged: https://github.com/wasmerio/wasmer/pull/4011

**Comment by kwonoj** [comment]
Yes, saw that but per https://github.com/wasmerio/wasmer/issues/3793#issuecomment-1607683339 it is still recommended to reuse if possible?

File: crates/swc_plugin_runner/src/wasix_runtime.rs [hunk 1]
```diff
@@ -1,7 +1,36 @@
 use std::{path::PathBuf, sync::Arc};
 
+use parking_lot::Mutex;
+use swc_common::sync::{Lazy, OnceCell};
+use wasmer::Store;
 use wasmer_wasix::Runtime;
 
+/// A shared instance to plugin runtime engine.
+/// ref: https://github.com/wasmerio/wasmer/issues/3793#issuecomment-1607117480
+static ENGINE: Lazy<OnceCell<Mutex<wasmer::Engine>>> = Lazy::new(|| {
```
Comment (kdy1) [inline crates/swc_plugin_runner/src/wasix_runtime.rs:10]: Why do we use `Lazy<OnceCell<T>>>`? Lazy contains `OnceCell` internally

Comment (kwonoj) [reply crates/swc_plugin_runner/src/wasix_runtime.rs]: probably a simple mistake while copy some codes.

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_plugin_runner

**Pull request merged**